src/lib.rs

```diff
@@ -58,18 +66,20 @@ impl<W, T> PistonWindow<W, T>
         use piston::event::Events;
         use piston::window::{ OpenGLWindow, Window };
 
-        let (mut device, mut factory) = gfx_device_gl::create(|s| window.borrow_mut().get_proc_address(s));
+        let mut device = gfx_device_gl::Device::new(|s| window.borrow_mut().get_proc_address(s));
+        let mut factory = device.spawn_factory();
```

Comment:
@kvark should this factory be stored within the PistonWindow as well, now that `Canvas` has been removed? Or should it no longer be necessary?